### PR TITLE
[WIP] Star in nested function

### PR DESCRIPTION
**Pull request by GumpacG**
### Description
WIP for * in nested function
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: core/src/main/java/org/opensearch/sql/ast/expression/NestedAllFields.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * Represent the All fields which is been used in SELECT *.
+ */
+@ToString
+@EqualsAndHashCode(callSuper = false)
+public class NestedAllFields extends UnresolvedExpression {
+  @Getter
+  private final String path;
+
+  public NestedAllFields(String path) {
+    this.path = path.split("[.][*]")[0];
```
Comment (Yury-Fridlyand) [inline core/src/main/java/org/opensearch/sql/ast/expression/NestedAllFields.java:27]:
```suggestion
    this.path = path.split("\\.")[0];
```

File: core/src/main/java/org/opensearch/sql/planner/physical/UnnestOperator.java [hunk 2]
```diff
@@ -67,6 +67,35 @@ public UnnestOperator(PhysicalPlan input, List<Map<String, ReferenceExpression>>
...
+    List<String> paths = fields.stream().map(path -> path.get("path")
+        .getAttr()).collect(Collectors.toList());
+
+    for (String path : paths) {
+      this.groupedPathsAndFields.put(path, new ArrayList<>());
+      for (String field : this.fields) {
+        if (field.contains(path)) {
+          this.groupedPathsAndFields.get(path).add(field);
+        }
+      }
```
Comment (Yury-Fridlyand) [inline core/src/main/java/org/opensearch/sql/planner/physical/UnnestOperator.java:95]:
```suggestion
      this.groupedPathsAndFields.put(path, 
          this.fields.stream().filter(f -> field.contains(path)).collect(Collectors.toList()));
```

File: sql/src/main/java/org/opensearch/sql/sql/parser/AstBuilder.java [hunk 3]
```diff
@@ -199,6 +202,11 @@ private UnresolvedExpression visitSelectItem(SelectElementContext ctx) {
     UnresolvedExpression expr = visitAstExpression(ctx.expression());
 
     if (expr instanceof Alias) {
+      List funcArgsParts = ((QualifiedName) ((Function) ((Alias) expr)
+          .getDelegated()).getFuncArgs().get(0)).getParts();
+      if (funcArgsParts.get(funcArgsParts.size() - 1).equals("*")) {
+        return new NestedAllFields(((Alias) expr).getName());
+      }
```
Comment (Yury-Fridlyand) [inline sql/src/main/java/org/opensearch/sql/sql/parser/AstBuilder.java:209]: Maybe you need to check first whether `delegated` is a function and validate function name.

Comment (GumpacG) [reply sql/src/main/java/org/opensearch/sql/sql/parser/AstBuilder.java]: Changed in https://github.com/Bit-Quill/opensearch-project-sql/pull/256/commits/14f410c441df9436a0d4605c063923224bb32be8

File: core/src/main/java/org/opensearch/sql/planner/physical/UnnestOperator.java (hunk 2 above)
Comment (GumpacG) [reply core/src/main/java/org/opensearch/sql/planner/physical/UnnestOperator.java]: Changed in https://github.com/Bit-Quill/opensearch-project-sql/pull/256/commits/14f410c441df9436a0d4605c063923224bb32be8

File: core/src/main/java/org/opensearch/sql/ast/expression/NestedAllFields.java (hunk 1 above)
Comment (GumpacG) [reply core/src/main/java/org/opensearch/sql/ast/expression/NestedAllFields.java]: Changed in https://github.com/Bit-Quill/opensearch-project-sql/pull/256/commits/14f410c441df9436a0d4605c063923224bb32be8

File: core/src/main/java/org/opensearch/sql/analysis/NestedAnalyzer.java [hunk 4]
```diff
@@ -43,6 +44,17 @@ public LogicalPlan visitAlias(Alias node, AnalysisContext context) {
     return node.getDelegated().accept(this, context);
   }
 
+
+  @Override
+  public LogicalPlan visitNestedAllFields(NestedAllFields node, AnalysisContext context) {
+    Map<String, ReferenceExpression> args;
+    args = Map.of(
```
Comment (MaxKsyunz) [inline core/src/main/java/org/opensearch/sql/analysis/NestedAnalyzer.java:51]: Can be combined.

**Comment by GumpacG** [comment]
Needs to be reworked. Will follow up with another PR.

**Pull request closed without merging**